server/pg/pg_comm_task.cpp

```diff
@@ -365,13 +365,19 @@ void PgSQLCommTaskBase::HandleClientPacket(std::string_view packet) {
 }
 
 void PgSQLCommTaskBase::DescribeAnalyzedQuery(
-  const query::Query& query, const std::vector<VarFormat>& formats,
+  const SqlStatement& statement, const std::vector<VarFormat>& formats,
   bool extended) {
+  const auto& query = *statement.query;
   const auto& output_type = query.GetOutputType();
   const uint16_t num_fields = output_type ? output_type->size() : 0;
+  const auto* pg_node = castNode(RawStmt, statement.tree.GetRoot());
+  SDB_ASSERT(pg_node);
+
   // If it's DML but name of column isn't "rows" it means it's explain for DML.
   // In such case we should send rows as usual.
-  if (num_fields == 0 || (query.IsDML() && output_type->nameOf(0) == "rows")) {
+  if ((pg_node->stmt->type == T_CallStmt) ||
+      (!query.IsDataQuery() && num_fields == 0) ||
+      (query.IsDML() && output_type->nameOf(0) == "rows")) {
```

Comment:
```
  if (!query.IsDataQuery() ||
      (pg_node->stmt->type == T_CallStmt) ||
      (query.IsDML() && output_type->nameOf(0) == "rows")) {
```

Reply:
that is not true, we have to check `num_fields == 0` in case of non data queries.
For example, `SHOW`